src/transformers/models/whisper/generation_whisper.py

```diff
@@ -195,7 +197,9 @@ def _extract_token_timestamps(self, generate_outputs, alignment_heads, time_prec
                 dim=2,
             )
 
-        timestamps = torch.zeros_like(generate_outputs.sequences, dtype=torch.float32)
+        # make sure timestamps are as long as weights
+        input_length = weight_length or cross_attentions[0].shape[2]
+        timestamps = torch.zeros_like(generate_outputs.sequences, dtype=torch.float32)[:, : input_length + 1]
```

Comment:
Now that we don't use `forced_decoder_ids` anymore for whisper generate -> we need to potentially cut the length of `timestamps` according to the actual length of the input